src/app/services/qhana-backend.service.ts

```diff
@@ -242,8 +242,13 @@ export class QhanaBackendService {
         return this.http.delete(`${this.rootUrl}/plugin-endpoints/${endpoint.endpointId}`).pipe(map(() => { return; }));
     }
 
-    public getExperimentsPage(page: number = 0, itemCount: number = 10): Observable<ApiObjectList<ExperimentApiObject>> {
-        return this.http.get<ApiObjectList<ExperimentApiObject>>(`${this.rootUrl}/experiments`);
+    public getExperimentsPage(page: number = 0, itemCount: number = 10, search: string | undefined = undefined, sort: number = 1): Observable<ApiObjectList<ExperimentApiObject>> {
+        let queryParams = new HttpParams();
+        if (search) {
+            queryParams = queryParams.append("search", search);
```

Comment:
is query `params.append` not mutating the original object?

Reply:
Nope, HttpParams is immutable.

Reply:
I did not check thy type thinking, that angular would surely use the JavaScript type <https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams>
If I have read the httpclient signature correctly params also accepts simple objects. But only do that if it improves readability...